cmd/ks-apiserver/app/server.go

```diff
@@ -120,14 +120,16 @@ func initializeServicemeshConfig(s *options.ServerRunOptions) {
 	kconfig.Set(config)
 
 	// Set kiali config
-	kubeconfig, err := clientcmd.BuildConfigFromFlags("", s.KubernetesOptions.KubeConfig)
-	if err != nil {
-		fmt.Println(err)
-	}
-	k8sClient, err := kubernetes.NewClientFromConfig(kubeconfig)
-	if err != nil {
-		fmt.Println(err)
+	if len(s.KubernetesOptions.KubeConfig) != 0 {
+		kubeconfig, err := clientcmd.BuildConfigFromFlags("", s.KubernetesOptions.KubeConfig)
+		if err != nil {
+			fmt.Println(err)
```

Comment:
If there is an error initialize kubeconfig, apiserver wouln't run normally, so exit.

Reply:
Yes, but it will first meet an error of kiali with unreadable error.

Reply:
A little bit confused here, what does that mean? Can you give an example.

Reply:
If the kubeconfig leaves blank, then the error is as follows:

```bash
github.com/kiali/kiali/kubernetes.NewClientFromConfig(0x0, 0x0, 0x0, 0x0)
	/home/roland/go/pkg/mod/github.com/kubesphere/kiali@v0.15.1-0.20201110082537-0c2b977257d4/kubernetes/client.go:174 +0xa3
kubesphere.io/kubesphere/cmd/ks-apiserver/app.initializeServicemeshConfig(0xc0007e26f0)
```

Reply:
So we don't want to display this raw error message back to the user, instead we can return formated message, for example
```
   fmt.Sprintf("create kubernetes client failed, please double check your kubeconfig file, %v", err)
```

Reply:
We cannot throw errors whenever the kubeconfig leaves blank. Kubeconfig is only needed when we debug locallly.

Reply:
OK, I see